src/app/components/elements/markup/markdownRendering.ts

```diff
@@ -79,10 +80,21 @@ export const renderInlineGlossaryTerms = (markdown: string) => {
 };
 
 // RegEx replacements to match Latex inspired Isaac Physics functionality
-export const regexProcessMarkdown = (markdown: string) => {
+export const regexProcessMarkdown = (pageContext?: PageContextState) => (markdown: string) => {
     const regexRules = {
         "[$1]($2)": /\\link{([^}]*)}{([^}]*)}/g,
     };
+    if (isPhy) {
+        let linkContext = "";
+        if (pageContext?.subject && isDefined(pageContext?.stage) && pageContext.stage.length === 1) {
+            linkContext = `/${pageContext.subject}/${pageContext.stage[0]}`;
+        }
+
+        Object.assign(regexRules, {
+            [`[**Glossary**](${linkContext}/glossary)`]: /\*\*Glossary\*\*/g,
+            [`[**Concepts**](${linkContext}/concepts)`]: /\*\*Concepts\*\*/g,
+        });
```

Comment:
```suggestion
         if (pageContext?.subject && !isSingleStageContext(pageContext)) {
            Object.assign(regexRules, {
                [`[**Glossary**](/glossary?subjects=${pageContext.subject})`]: /\*\*Glossary\*\*/g,
                [`[**Concepts**](/concepts?types=${pageContext.subject})`]: /\*\*Concepts\*\*/g,
            });
        }
        else {
            let linkContext = "";
            
            if (pageContext?.subject && isSingleStageContext(pageContext)) {
                linkContext = `/${pageContext.subject}/${pageContext.stage![0]}`;
            }

            Object.assign(regexRules, {
                [`[**Glossary**](${linkContext}/glossary)`]: /\*\*Glossary\*\*/g,
                [`[**Concepts**](${linkContext}/concepts)`]: /\*\*Concepts\*\*/g,
            });
        }
```
This works nicely for questions with single-stage contexts, but for multi-stage questions (e.g. `/questions/aircraft_balance` ) this links to the unfiltered generic concepts/glossary, which as you pointed out is the reason why these links were removed in the first place. Something like this would at least filter the concepts/terms to the relevant subject?